package-with/pam-rhel.conf

```diff
@@ -1,3 +1,11 @@
-auth		include	password-auth
-account		include	password-auth
-session		include	password-auth
+auth		substack	password-auth
+auth   		include		postlogin
+account		required	pam_sepermit.so
+account		include		password-auth
+session		required	pam_selinux.so close
+session		required 	pam_loginuid.so
+session		required	pam_selinux.so open env_params
+session		required	pam_namespace.so
+session		optional	pam_keyinit.so force revoke
+session		include		password-auth
+session		include		postlogin
```

Comment:
It would be good to get an explanation of these changes.  The Redhat PAM file was provided by another user so I freely admit it might need securing.  Have these been copied from another service?  Does Redhat not have standard include files for this purpose?

Reply:
I copied them from sshd, omitting the `password` entries which do not apply and also `pam_motd` which doesn't make sense for a non-shell.  They all appears to make sense, though some are a little unclear.

There are include files for password-auth and postlogin, but nothing for this whole sequence.  Here's what proftpd does:
```
session    optional     pam_keyinit.so force revoke
auth       required     pam_listfile.so item=user sense=deny file=/etc/ftpusers onerr=succeed
auth       required     pam_shells.so
auth       include      password-auth
account    include      password-auth
session    required     pam_loginuid.so
session    include      password-auth
```
vsftpd is basically the same.  Maybe that would be a better starting point?

Reply:
Might be good to add this:
```
auth            required        pam_succeed_if.so user != root quiet_success
```
This will disallow root access.  sshd by default does not allow root logins on Fedora, and most other distros too, so it seems appropriate.